src/flake8_tidy_imports/__init__.py

```diff
@@ -63,7 +70,25 @@ def parse_options(cls, options: Any) -> None:
             module, message = line.split("=", 1)
             module = module.strip()
             message = message.strip()
-            cls.banned_modules[module] = message
+
+            if "*" in module[:-1] or module == "*":
+                # unstructured
+                cls.banned_unstructured_patterns.append(
+                    (cls.compile_unstructured_glob(module), message)
+                )
+            elif module.endswith(".*"):
+                # structured
+                cls.banned_structured_patterns.append((module, message))
+                # Also check for exact matches without the wilcard
+                # e.g. "foo.*" matches "foo"
+                prefix = module[:-2]
+                if prefix not in cls.banned_modules:
```

Comment:
This is so that the message for a "concrete" pattern takes precedent; I could also resolve this by processing the structured wildcards first and letting the concrete ones over overwrite (which is the approach mypy takes).